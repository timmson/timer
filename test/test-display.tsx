import React from "react"
import {render, screen} from "@testing-library/react"
import user from "@testing-library/user-event"
import "@testing-library/jest-dom"
import "regenerator-runtime/runtime"

import Display from "../src/display"
import Context from "../src/context"

describe("Display should", () => {

	test("return stopped timer", () => {
		render(<Display status={"STOPPED"} value={60}/>)

		expect(screen.getByText(/01/)).toBeInTheDocument()
		expect(screen.getByText(/:/)).toBeInTheDocument()
		expect(screen.getByText(/00/)).toBeInTheDocument()
	})

	test("return started timer", () => {
		render(<Display status={"STARTED"} value={60}/>)

		expect(screen.getByText(/01/)).toBeInTheDocument()
		expect(screen.getByText(/:/)).toBeInTheDocument()
		expect(screen.getByText(/00/)).toBeInTheDocument()
	})

	test("return alerted timer", () => {
		render(<Display status={"ALERTED"} value={0}/>)

		expect(screen.getByText(/0/)).toBeInTheDocument()
	})

	test("give possibility to click", async () => {
		const expectedAction = {type: "ACTION_TOGGLE"}

		render(
			<Context.Provider value={(action) => {
				expect(action).toEqual(expectedAction)
			}}>
				<Display status={"ALERTED"} value={0}/>
			</Context.Provider>
		)

		await user.click(screen.getByText(/:/).parentElement)

		expect.assertions(1)
	})

})
