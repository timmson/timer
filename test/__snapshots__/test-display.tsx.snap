// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should return alerted timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer alert"
      onClick={[Function]}
    >
      00
      <span
        style={{}}
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;

exports[`Display should return started timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer normal"
      onClick={[Function]}
    >
      01
      <span
        style={
          {
            "animation": "blinker 1s linear infinite",
          }
        }
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;

exports[`Display should return stopped timer 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="timer normal"
      onClick={[Function]}
    >
      01
      <span
        style={{}}
      >
        :
      </span>
      00
    </div>
  </div>
</div>
`;
